filebeat/prospector/prospector.go

```diff
@@ -105,6 +105,10 @@ func (p *Prospector) Run() {
 				logp.Info("Prospector channel stopped")
 				return
 			case event := <-p.harvesterChan:
+				// Add ttl if cleanOlder is enabled
+				if p.config.CleanOlder != 0 {
+					event.FileState.Ttl = int64(p.config.CleanOlder.Seconds())
```

Comment:
It is used to update the state (not cleanup) on line 117 below. Update already has a mutex operation. Is that enough?